src/coreclr/jit/emitxarch.cpp

```diff
@@ -5971,6 +5998,15 @@ bool emitter::HasSideEffect(instruction ins, emitAttr size)
         }
 #endif // TARGET_AMD64
 
+        case INS_kmovb:
+        case INS_kmovw:
+        case INS_kmovd:
+        case INS_kmovq:
+        {
+            hasSideEffect = true;
```

Comment:
This should be "false" for `kmovq` since its a 64-bit to 64-bit move and therefore doesn't zero extend, right?

Reply:
This is notably still unresolved